docs/introduction.md

```diff
@@ -0,0 +1,138 @@
+
+### Environment Setup
+
+Optionally, if you want to setup a local development environment, [you can follow our guide to setting up your environment on your local machine](getting-started). Our you can follow along with the interactive examples used throughout this guide.
+
+## Interactive examples
+
+```SnackPlayer name=Hello%20World
+import React from 'react';
+import { Text, View } from 'react-native';
+
+export default function YourApp() {
+  return (
+    <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
+      <Text>Hello, world!</Text>
+    </View>
+  );
+}
+```
+
+The above is a Snack Player. It’s [a handy tool created by Expo](https://expo.io/tools#snack) to embed and run React Native projects on and share how they render in platforms like Android and iOS. We will be using them a lot to demonstrate code samples throughout these docs. You can play around with sample code provided in the Snack, or you can [set up an environment on your local machine](getting-started) and paste the code into your `App.js` file there.
+
+## Function Components and Class Components
+
+With React, you can make components using either classes or functions. Originally, class components were the only components that could have state. However, [React 16.8 introduced an API called “Hooks” for adding state and more to function components](https://reactjs.org/docs/hooks-intro.html). In March 2019, [React Native 0.58 adopted Hooks as well](https://facebook.github.io/react-native/blog/2019/03/12/releasing-react-native-059).
+
+Because Hooks are the future-facing way to write your React components, we wrote this introduction using function component examples. Where useful, we also cover class components under a toggle:
+
+<div class="toggler">
+  <ul role="tablist" class="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTabs('syntax', 'functional')">
+      Function Component Example
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTabs('syntax', 'classical')">
+      Class Component Example
+    </li>
+  </ul>
+</div>
+
+<block class="functional syntax" />
+
+```SnackPlayer name=Hello%20World%20Function%20Component
+import React from 'react';
+import { Text, View } from 'react-native';
+
+export default function YourApp() {
+  return (
+    <View style={{
+        flex: 1,
+        justifyContent: 'center',
+        alignItems: 'center'
+      }}>
+      <Text>Hello, world!</Text>
+    </View>
+  );
+}
+```
+
+<block class="classical syntax" />
+
+```SnackPlayer name=Hello%20World%20Class%20Component
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+
+export default class HelloWorldApp extends Component {
```

Comment:
Nit: this one is `HelloWorldApp` but the other one is `YourApp`. Use the same name?

Reply:
Sloppy!